src/analyses/base.ml

```diff
@@ -815,23 +815,23 @@ struct
 
   (* interpreter end *)
 
-  let query ctx (q:Q.t) =
+  let query ctx (type a) (q: a Q.t): a Q.result =
     let to_int = BI.to_int64 in
     match q with
     | Q.EvalFunvar e ->
       begin
         let fs = eval_funvar ctx e in
         (*          Messages.report ("Base: I should know it! "^string_of_int (List.length fs));*)
-        `LvalSet (List.fold_left (fun xs v -> Q.LS.add (v,`NoOffset) xs) (Q.LS.empty ()) fs)
+        LvalSet (List.fold_left (fun xs v -> Q.LS.add (v,`NoOffset) xs) (Q.LS.empty ()) fs)
       end
     | Q.EvalInt e -> begin
-        match eval_rv ctx.ask ctx.global ctx.local e with
```

Comment:
Continuing on my type-safety mania, I tried it here: https://github.com/goblint/analyzer/commit/fc109df09fd3c0c04ae2689d277536b1544c4e99. It's horrible because so many [coercions](https://ocaml.org/releases/4.12/htmlman/expr.html#ss:expr-coercions) to typecheck because apparently OCaml hates if you pass a more precise polymorphic variant type than the return type annotation...

Reply:
I would think so, but we can leave that as a separate issue if anyone wants to try.